src/core/entities/post.entity.ts

```diff
@@ -0,0 +1,38 @@
+import "reflect-metadata";
+
+import { Field, Int, ObjectType } from "type-graphql";
+
+import {
+  //   BooleanProfileField,
+  //   NumberProfileField,
+  //   ProfilePrivacyField,
+  //   ProfilePrivacyFieldInput,
+  StringProfileField,
+} from "./profile.entity";
+
+@ObjectType()
+export class Post {
+  @Field(() => Int)
+  id!: number;
+
+  @Field()
+  userId?: number;
+
+  @Field(() => StringProfileField, { nullable: true })
```

Comment:
We don't need to use the StringProfileField here, although I can totally understand the line of thinking!! The reason we needed this in profile was because we were trying to store 2 bits of info for each field. For the post, the visibility for privacy can just be its own field, and we can use the enum!! So content can just be a string.